jest.setup.js

```diff
@@ -24,3 +24,43 @@ jest.mock('react-i18next', () => ({
     init: jest.fn(),
   },
 }));
+
+// Provide mock implementations for Firebase App and Messaging modules to simulate Firebase behavior in tests
```

Comment:
See - [How to write better code comments](https://dev.to/adammc331/todo-write-a-better-comment-4c8c)